ocaml/typing/mode.ml

```diff
@@ -1335,68 +1341,19 @@ let locality_as_regionality m =
 let regional_to_global m =
   S.Positive.via_monotone Locality.Obj.obj C.Regional_to_global m
 
-module Comonadic_with_regionality = struct
-  module Const = C.Comonadic_with_regionality
-
-  module Obj = struct
-    type const = Const.t
-
-    module Solver = S.Positive
-
-    let obj = C.Comonadic_with_regionality
-  end
-
-  include Common (Obj)
-
-  type error = Error : (Const.t, 'a) C.Axis.t * 'a Solver.error -> error
-
-  type equate_error = equate_step * error
-
-  open Obj
-
-  let proj_obj ax = C.proj_obj ax obj
-
-  let proj ax m = Solver.via_monotone (proj_obj ax) (Proj (Obj.obj, ax)) m
-
-  let meet_const c m = Solver.via_monotone obj (Meet_with c) m
-
-  let join_const c m = Solver.via_monotone obj (Join_with c) m
-
-  let min_with ax m =
-    Solver.via_monotone Obj.obj (Min_with ax) (Solver.disallow_right m)
-
-  let max_with ax m =
-    Solver.via_monotone Obj.obj (Max_with ax) (Solver.disallow_left m)
-
-  let join_with ax c m = join_const (C.min_with Obj.obj ax c) m
+module type Areality = sig
+  module Const : C.Areality
 
-  let meet_with ax c m = meet_const (C.max_with Obj.obj ax c) m
+  module Obj : Obj with type const = Const.t and module Solver = S.Positive
 
-  let imply c m = Solver.via_monotone obj (Imply c) (Solver.disallow_left m)
+  val comonadic_with_areality : Const.t C.comonadic_with C.obj
```

Comment:
Instead of defining this, I think it's more canonical to use the `comonadic_with_obj` function.